pkg/sql/schemachanger/eav/values.go

```diff
@@ -0,0 +1,83 @@
+// Copyright 2021 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package eav
+
+import "sync"
+
+// Values is used to store a select elements from a Tree.
+type Values map[Ordinal]Value
+
+// GetValues retrieves a Values instance from the sync pool. Use
+// Release to put it back in the pool. Values are often used in
+// contexts with well-defined lifecycles, hence the pooling.
+func GetValues() Values {
+	return valuesSyncPool.Get().(Values)
+}
+
+// Copy clones the Values into a newly allocated map.
+func (vv Values) Copy() Values {
+	cpy := GetValues()
+	for k, v := range vv {
+		cpy[k] = v
+	}
+	return cpy
+}
+
+// Release releases the Values back into the pool.
+func (vv *Values) Release() {
+	v := *vv
+	for k := range v {
```

Comment:
Are these maps going to get big? I know sync pools get dropped during memory pressure, but is this memory pooling and associated keeping-around of map space premature optimization?